packages/core/src/components/editable-text/editableText.tsx

```diff
@@ -357,7 +357,7 @@ export class EditableText extends AbstractPureComponent<EditableTextProps, Edita
         }
 
         const hasModifierKey = altKey || ctrlKey || metaKey || shiftKey;
-        if (event.key === "Enter") {
+        if (event.key === "Enter" && !event.nativeEvent.isComposing) {
```

Comment:
I think we actually need to do this earlier, before we handle escape. Normally, at least on macOS, pressing escape while composing cancels the _current_ sequence. Both with and without this change, it cancels the entire edit, i.e. everything that was changed will be abandoned, instead of just the current sequence discarded and other changes and editing mode preserved.

Reply:
Hey you're right that *maybe* we should not escape when the IME is composing, but in my understanding people don't usually press escape during typing. Personally, I would expect to escape even when composing